app/controllers/registration/applicant/CaptureTelephoneNumberController.scala

```diff
@@ -47,16 +47,19 @@ class CaptureTelephoneNumberController @Inject()(view: capture_telephone_number,
         for {
           applicant <- applicantDetailsService.getApplicantDetails
           filledForm = applicant.telephoneNumber.fold(TelephoneNumberForm.form)(tn => TelephoneNumberForm.form.fill(tn.telephone))
-        } yield
-          Ok(view(routes.CaptureTelephoneNumberController.submit, filledForm))
+          isTransactor <- vatRegistrationService.isTransactor
+          name <- if (isTransactor) applicantDetailsService.getTransactorApplicantName else Future.successful(None)
+        } yield Ok(view(routes.CaptureTelephoneNumberController.submit, filledForm, name))
   }
 
   def submit: Action[AnyContent] = isAuthenticatedWithProfile() {
     implicit request =>
       implicit profile =>
         TelephoneNumberForm.form.bindFromRequest().fold(
           formWithErrors =>
-            Future.successful(BadRequest(view(routes.CaptureTelephoneNumberController.submit, formWithErrors))),
+            applicantDetailsService.getTransactorApplicantName.flatMap { name =>
```

Comment:
You don't need to flatMap here - you can tell because you're having to wrap the response in Future.successful() - just use map. Same for the rest of the BadRequest bits